src/natcap/invest/utils.py

```diff
@@ -539,3 +553,163 @@ def mean_pixel_size_and_area(pixel_size_tuple):
+
+    Returns:
+        An OSR Coordinate Transformation object
+
+    """
+    # Make a copy of the base and target spatial references to avoid side 
+    # effects from mutation of setting the axis mapping strategy
+    base_ref_wkt = base_ref.ExportToWkt()
+    target_ref_wkt = target_ref.ExportToWkt()
+
+    base_ref_copy = osr.SpatialReference()
+    target_ref_copy = osr.SpatialReference()
+    
+    base_ref_copy.ImportFromWkt(base_ref_wkt)
+    target_ref_copy.ImportFromWkt(target_ref_wkt)
+    
+    base_ref_copy.SetAxisMappingStrategy(osr_axis_mapping_strategy)
+    target_ref_copy.SetAxisMappingStrategy(osr_axis_mapping_strategy)
+
+    transformer = osr.CreateCoordinateTransformation(
+        base_ref_copy, target_ref_copy)
+    return transformer
+
+
+def _assert_vectors_equal(
+        expected_vector_path, actual_vector_path, field_value_atol=1e-3):
+    """Assert two vectors are equal.
+
+    Assert spatial reference, feature count, geometries, field names, and
+    values are equal with no respect to order of field names or geometries.
+
+    Args:
+        actual_vector_path (string): path on disk to a gdal Vector dataset.
+        expected_vector_path (string): path on disk to a gdal Vector dataset
+            to use as the ground truth.
+        field_value_atol (float): the absolute tolerance for comparing field
+            attribute values, default=1e-3.
+
+    Returns:
+        A tuple (bool, string) where the boolean element is True if vectors
+        are equal and False if not. The string element is a message of
+        "success" if True or detailed error message if False.
+    """
+    try:
+        # Open vectors
+        actual_vector = gdal.OpenEx(actual_vector_path, gdal.OF_VECTOR)
+        actual_layer = actual_vector.GetLayer()
+        expected_vector = gdal.OpenEx(expected_vector_path, gdal.OF_VECTOR)
+        expected_layer = expected_vector.GetLayer()
+
+        # Check projections
+        expected_projection = expected_layer.GetSpatialRef()
+        expected_projection_wkt = expected_projection.ExportToWkt()
+        actual_projection = actual_layer.GetSpatialRef()
+        actual_projection_wkt = actual_projection.ExportToWkt()
+        if expected_projection_wkt != actual_projection_wkt:
+            return (
+                False,
+                "Vector projections are not the same. \n"
+                f"Expected projection wkt: {expected_projection_wkt}. \n"
+                f"Actual projection wkt: {actual_projection_wkt}. ")
+
+        # Check feature count
+        actual_feat_count = actual_layer.GetFeatureCount()
+        expected_feat_count = expected_layer.GetFeatureCount()
+        if expected_feat_count != actual_feat_count:
+            return (
+                False,
+                "Vector feature counts are not the same. \n"
+                f"Expected feature count: {expected_feat_count}. \n"
+                f"Actual feature count: {actual_feat_count}. ")
```

Comment:
Hey @dcdenu4 I'm not sure I'm interpreting this correctly.  It looks like this is returning the validity and error message, which looks inconsistent with the way that it's being used in some of the tests.  In `tests/test_fisheries.py` (line 261), for example (other files use this as well), we have:
```python
utils._assert_vectors_equal(
    aoi_path,
    os.path.join(out_workspace_dir, 'test_aoi_for_subset.shp'))
```

If there were an error in the vector comparison, would the `False` returned be captured and interpreted as a test error?  I know we used to explicitly raise `AssertionError` ... is that no longer required to indicate a test failure?

Reply:
Holy toledos. Good call. Since the 'assertion' function was living inside `utils` I wanted to be able to provide better and more informative failure messages. I thought I had updated all the tests to reflect how the function returned... 

But now that you mention it, perhaps raising `AssertionError` with a message is more descriptive of what the function wants to be.

Reply:
Fixed here https://github.com/natcap/invest/pull/150/commits/ced2e8f4f3cfebcddde923d0a878d8d43ff3b1ae